modules/proxy/mod_proxy_fcgi.c

```diff
@@ -348,10 +360,11 @@ static apr_status_t send_environment(proxy_conn_rec *conn, request_rec *r,
     fcgi_req_config_t *rconf = ap_get_module_config(r->request_config, &proxy_fcgi_module);
     fcgi_dirconf_t *dconf = ap_get_module_config(r->per_dir_config, &proxy_fcgi_module);
 
-    if (rconf) {
-       if (rconf->need_dirwalk) {
-          ap_directory_walk(r);
-       }
+    if (rconf && rconf->need_dirwalk) {
+        char *saved_filename = r->filename;
+        r->filename = r->uri;
+        ap_directory_walk(r);
```

Comment:
See https://lists.apache.org/thread/4gw6tsp0xlpdf3l9gtz1jnr5wzycsfg8:

Why not using the result of the below Strip proxy: prefixes for `r->filename`? In case of a `Rewriterule` I guess `r->uri` and `r->filename` could be fundamentally different path wise.

Reply:
Maybe it helps to say a little about this obscure option to try to calculate PATH_INFO. It's meant/doc'ed for ProxyPass/ProxyPassMatch configurations where there was no mapping to FS hence no split off for PATH_INFO.  

I don't think any of these proxy-fcgi-pathinfo  options are widely used.  I guess if r->filename had proxy: prefix stuff in it this was maybe useless since inception hence no users?

OTOH like you say, just peeling it all off as we will already do below would seem to give a better result assuming the user doesn't intend to keep some "original" PATH_INFO.

Reply:
I'll create another PR for this because this has nothing to do with BZ-69203 afterall..

Reply:
See #489